pkg/subctl/cmd/join.go

```diff
@@ -419,6 +444,41 @@ func isOpenShiftCVOEnabled(config *rest.Config) (bool, error) {
 	return scale.Spec.Replicas > 0, nil
 }
 
+func cidrIsPreConfigured(clusterID string, globalNetworks map[string]*globalnet.GlobalNetwork) bool {
+	// GlobalCIDR is not pre-configured
+	if globalNetworks[clusterID] == nil || globalNetworks[clusterID].GlobalCIDRs == nil || len(globalNetworks[clusterID].GlobalCIDRs) <= 0 {
+		return false
+	}
+	// GlobalCIDR is pre-configured
+	return true
+}
+
+func validateGlobalnetConfiguration(subctlData *datafile.SubctlData) {
+	if subctlData.GlobalnetCidrRange != "" && globalnetClusterSize != 0 && globalnetClusterSize != subctlData.GlobalnetClusterSize {
+		clusterSize, err := globalnet.GetValidClusterSize(subctlData.GlobalnetCidrRange, globalnetClusterSize)
+		if err != nil || clusterSize == 0 {
+			exitOnError("Invalid globalnet-cluster-size", err)
+		}
+		subctlData.GlobalnetClusterSize = clusterSize
+	}
+
+	if globalnetCIDR != "" && globalnetClusterSize != 0 {
+		err := errors.New("Both globalnet-cluster-size and globalnet-cidr can't be specified. Specify either one\n")
+		exitOnError("Invalid configuration:", err)
+	}
+
+	if globalnetCIDR != "" {
+		_, _, err := net.ParseCIDR(globalnetCIDR)
+		exitOnError("Specified globalnet-cidr is invalid", err)
+	}
+
+	if subctlData.GlobalnetCidrRange == "" && (globalnetCIDR != "" || globalnetClusterSize != 0) {
+		fmt.Printf("* globalnet is not enabled in broker. Ignoring specified globalnet-cidr or globalnet-cluster-size\n")
```

Comment:
Do we get fmt.Printf messages when we run subctl? 
Should we actually use status.QueueSuccessMessage here?

Reply:
yes we get printf output

Reply:
eg: fmt.Printf("* Discovering network details\n")

Reply:
Yes, this should use either `status.QueueSuccessMessage` or `status.QueueFailureMessage`, depending on the desired format.

Reply:
thanks @skitt 

I'd prefer to use status.QueueSuccessMessage here as its not a failure. If we show it as a failure, user might think that something is broken as part of the deployment. @vthapar your comments on this?

Reply:
Done